### PR TITLE
Feature/omega 27 customer relation overview page

### PR DESCRIPTION
**Pull request by ragnhildlarsen**
OMEGA-27

## What is the reason for this change?
Customers that have multiple customer relations need an overview page for all their customer relations.  

### What is the expected result?
Customers with multiple customer relations should se an overview page if going to /minside/kundeforhold. Customers with only one customer relation should be redirected to the only customer relation they have /minside/kundeforhold/customerId. Customers with no customer relations should be redirected to access request page /minside/accessrequest. When you click on a custom relation card on the overview page you should be navigated to a detail page with an url /minside/kundeforhold/customerId which currently only shows "test".

[Figma sketches](https://www.figma.com/file/sJHoScmQRq0hlr6AcHI9WB/Elvia.no?node-id=9334%3A57402)

## How is it solved?
A new template is created for CustomerRelationOverview. This component accesses the customer store and uses that data to decide what should be shown or if the customer should be redirected. 
A dummy template for CustomerDetail have been created to be a placeholder for the detail page for a Customer Relation (this might be excessive if someone else is working on the actual page). 

## Any changes/updates that may affect others?
|                                     |Changed|
|-------------------------------------|-------|
| Unit tests                          |      |
| Dependencies                        |       |
| Environment variables               |       |
| Build scripts                       |       |

## How can it be tested and verified?

- [ ] Test behaviour for different test customers and check that the correct data is shown.
- [ ] UI looks and behaves identical to design defined in Figma.
- [ ] enter /minside/kundeforhold in the url to navigate to the page
- [ ] test user Per Hansen with birth number 05039703930 has 1 private and 1 organization customer relation
- [ ] test user Ole UtenKontrakt with birth number 23028791364 has no customer relations
- [ ] test user Testprivat122 with birth number 23025139593 has 1 customer relation
- [ ] functionality is only visible in development environment

### CODE REVIEW

**Comment by OysteinThuen** [comment]
Build #3007 successfully built and deployed.

- Prod: https://5f4df6a849ba2fc64d8572c3--elvia-prod.netlify.app
- Test: https://5f4df69ef0854f9e7f72d17f--elvia-test.netlify.app

**Comment by OysteinThuen** [comment]
Build #3012 successfully built and deployed.

- Prod: https://5f4e03f949ba2fd8908573f9--elvia-prod.netlify.app
- Test: https://5f4e043890bc06a8916cb5ba--elvia-test.netlify.app

File: packages/web/src/components/CustomerCard/CustomerCard.module.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
+@import '../../theme';
+
+.relationCard {
+  border-radius: emCalc(8px);
+  box-shadow: 0px 0px 40px rgba(0, 0, 0, 0.03);
```
Comment (eTallang) [inline packages/web/src/components/CustomerCard/CustomerCard.module.scss:5]: I believe this shadow color exists as a variable in theme.

File: packages/web/src/components/CustomerCard/CustomerCard.module.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  background-color: white;
```
Comment (eTallang) [inline packages/web/src/components/CustomerCard/CustomerCard.module.scss:6]: So does the white color `map-get($colors, 'white')`

File: packages/web/src/components/CustomerCard/CustomerCard.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+} from '@hdd/index'
+import { getTerm, Terms } from '@utils/getTerm'
+import { Customer } from '@services/customer-portal-api'
+import * as css from './CustomerCard.module.scss'
+
+interface CustomerCardProps {
+  customer: Customer
+}
+
+export const CustomerCard = ({ customer }: CustomerCardProps) => {
```
Comment (eTallang) [inline packages/web/src/components/CustomerCard/CustomerCard.tsx:19]: To get typings for a functional react component, you can define the type like this:
```
export const CustomerCard: React.FC<CustomerCardProps> = ({customer}) = {
```

File: packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import { TitleLarge, TextCaps } from '@hdd/index'
+import { Terms, getTerm } from '@utils/getTerm'
+import { CustomerCard } from '@components/CustomerCard'
+import { Customer } from '@services/customer-portal-api'
+
+export const CustomerRelationOverview: React.FC = () => {
+  const privateCustomerRelations: Customer[] = []
+  const companyCustomerRelations: Customer[] = []
+
+  const customerRelations = useSelector(customersSelectors.getAllCustomers)
```
Comment (eTallang) [inline packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx:15]: Could we use some types on the `useSelector`-method to improve type safety?

**Comment by OysteinThuen** [comment]
Build #3065 successfully built and deployed.

- Prod: https://5f4e313311d0c4dc7846480c--elvia-prod.netlify.app
- Test: https://5f4e31169d1fbe00a859a83d--elvia-test.netlify.app

File: packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx (hunk 4 above)
Comment (ragnhildlarsen) [reply packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx:15]: Do you mean in this file or in the selectors file for customersSelectors? I see that some of the other selectors have type so I can add it in the selectors file like this:
`export const getAllCustomers = (state: RootState): Record<string, Customer> =>
  state.customers.customersById`

**Comment by OysteinThuen** [comment]
Build #3082 successfully built and deployed.

- Prod: https://5f4e39e56bfe1500c6cd8728--elvia-prod.netlify.app
- Test: https://5f4e39cd6bfe150127cd85c1--elvia-test.netlify.app

**Review by JonatanPaalsson — CHANGES_REQUESTED** [review]
Får ikke testet dette i test-miljö, kanske vi mangler node_env der? 🤔

File: packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    } else {
+      companyCustomerRelations.push(customerRelations[key])
+    }
+  })
+
+  const CustomerRelationOverviewLayout: React.FC = props => (
+    <>{props.children}</>
+  )
+
+  switch (Object.keys(customerRelations).length) {
```
[the remark this answers is not in the document]
Comment (ragnhildlarsen) [reply packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx:29]: I don't think it is necessary to factor in `privateCustomerRelations` in the switch, as the main goal of the switch is to separate between the different scenarios where you have 0, 1 or multiple customer relations. The separation between private and organisation customer relations is taken into account in the default case in the switch.

File: packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  Object.keys(customerRelations).forEach(key => {
+    if (customerRelations[key].type === 'Person') {
+      privateCustomerRelations.push(customerRelations[key])
+    } else {
+      companyCustomerRelations.push(customerRelations[key])
+    }
+  })
+
+  const CustomerRelationOverviewLayout: React.FC = props => (
+    <>{props.children}</>
```
[the remark this answers is not in the document]
Comment (ragnhildlarsen) [reply packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx:26]: I don' know. When I remove the `<>` and `</>` that wraps the props I get an error. Do you have a suggestion to how I can do it differently?

File: packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (JonatanPaalsson) [reply packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx:29]: Ah right, now I understand! I think `allCustomerRelations` would be a better name for `customerRelations`, but that is just preference 👍

**Review by JonatanPaalsson — APPROVED** [review]

**Pull request merged**